From: Felipe Sateler <fsateler@users.noreply.github.com>
Date: Mon, 28 Aug 2017 13:49:03 -0300
Subject: shared: Add a linker script so that all functions are tagget
 @SD_SHARED instead of @Base (#6669)

This helps prevent symbol collisions with other programs and libraries. In particular,
because PAM modules are loaded into the process that is creating the session, and
systemd creates PAM sessions, the potential for collisions is high.

Disambiguate all systemd calls by tagging a 'version' SD_SHARED.

Fixes #6624

(backported from commit 4f5993c3f48d6f5454f72aeb10b4642fde758076)
---
 Makefile.am              | 1 +
 src/shared/libshared.sym | 3 +++
 2 files changed, 4 insertions(+)
 create mode 100644 src/shared/libshared.sym

diff --git a/Makefile.am b/Makefile.am
index 13bbf21..5016d3b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1117,6 +1117,7 @@ libsystemd_shared_la_LIBADD = \
 
 libsystemd_shared_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
+	-Wl,--version-script=$(top_srcdir)/src/shared/libshared.sym \
 	-release $(PACKAGE_VERSION)
 
 
diff --git a/src/shared/libshared.sym b/src/shared/libshared.sym
new file mode 100644
index 0000000..e4ae17e
--- /dev/null
+++ b/src/shared/libshared.sym
@@ -0,0 +1,3 @@
+SD_SHARED {
+	global: *;
+};
